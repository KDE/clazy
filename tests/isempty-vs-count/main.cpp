#include <QtCore/QList>
#include <QtCore/QMap>
#include <QtCore/QMultiMap>

QList<int> getList() {return {};}

void test()
{
    QList<int> list;
    if (list.count()) {}   // Warning
    if (!list.count()) {}   // Warning
    bool b1 = list.count(); // Warning
    bool b2 = list.size(); // Warning
    if (list.indexOf(1)) {} // OK

    QMap<int, int> map;
    if (map.count()) {} // Warning
    if (map.count(2)) {} // Warning

    QMultiMap<int, int> multiMap;
    if (multiMap.count()) {} // Warning
    if (multiMap.count(2)) {} // Warning
    if (multiMap.count(2, 3)) {} // Warning

    if (getList().count()) {} // Warning
}

