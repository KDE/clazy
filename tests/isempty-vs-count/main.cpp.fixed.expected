#include <QtCore/QList>
#include <QtCore/QMap>
#include <QtCore/QMultiMap>

QList<int> getList() {return {};}

void test()
{
    QList<int> list;
    if (!list.isEmpty()) {}   // Warning
    if (list.isEmpty()) {}   // Warning
    bool b1 = !list.isEmpty(); // Warning
    bool b2 = !list.isEmpty(); // Warning
    if (list.indexOf(1)) {} // OK

    QMap<int, int> map;
    if (!map.isEmpty()) {} // Warning
    if (map.count(2)) {} // Warning

    QMultiMap<int, int> multiMap;
    if (!multiMap.isEmpty()) {} // Warning
    if (multiMap.count(2)) {} // Warning
    if (multiMap.count(2, 3)) {} // Warning

    if (!getList().isEmpty()) {} // Warning
}

